e2e/e2e-suite_test.go

```diff
@@ -90,3 +116,87 @@ func dump(kubeconfig, artifactsSuffix string) {
 	output, err := cmd.CombinedOutput()
 	Expect(err).ToNot(HaveOccurred(), string(output))
 }
+
+func startPortForward(namespace string) error {
+	name, err := findControlPlaneVMIName(namespace)
+	if err != nil {
+		klog.Errorf("error finding control plane vmi in namespace %s, %v\n", namespace, err)
+		return err
+	}
+	tenantApiPort, err = getFreePort()
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Found free port: %d\n", tenantApiPort)
+	cmd := CreateVirtctlCommand(context.Background(), "port-forward", "-n", namespace, fmt.Sprintf("vm/%s", name), fmt.Sprintf("%d:6443", tenantApiPort))
+	cancelFunc = cmd.Cancel
+	if cancelFunc == nil {
+		klog.Errorln("unable to get cancel function for port-forward command")
+		return nil
+	}
+
+	go func() {
+		out, err := cmd.CombinedOutput()
+		if err != nil {
+			klog.Errorf("error port-forwarding vmi %s/%s: %v, [%s]\n", namespace, name, err, string(out))
+			return
+		}
+	}()
+	return nil
+}
+
+func findControlPlaneVMIName(namespace string) (string, error) {
+	clientConfig := defaultInfraClientConfig(&pflag.FlagSet{})
+	cfg, err := clientConfig.ClientConfig()
+	Expect(err).ToNot(HaveOccurred())
+	virtClient = kubecli.NewForConfigOrDie(cfg)
+	Expect(err).ToNot(HaveOccurred())
+	vmiList, err := virtClient.KubevirtV1().VirtualMachineInstances(namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	var chosenVMI *kubevirtv1.VirtualMachineInstance
+	for _, vmi := range vmiList.Items {
+		if strings.Contains(vmi.Name, "-control-plane") {
+			chosenVMI = &vmi
+			break
+		}
+	}
+	if chosenVMI == nil {
+		return "", fmt.Errorf("Couldn't find controlplane vmi in namespace %s", namespace)
+	}
+	return chosenVMI.Name, nil
+}
+
+func getFreePort() (port int, err error) {
+	var a *net.TCPAddr
+	if a, err = net.ResolveTCPAddr("tcp", "localhost:0"); err == nil {
+		var l *net.TCPListener
+		if l, err = net.ListenTCP("tcp", a); err == nil {
+			defer l.Close()
+			return l.Addr().(*net.TCPAddr).Port, nil
+		}
+	}
+	return
+}
+
+func RunVirtctlCommand(ctx context.Context, args ...string) (string, error) {
+	cmd := CreateVirtctlCommand(ctx, args...)
+	outBytes, err := cmd.CombinedOutput()
+
+	return string(outBytes), err
+}
```

Comment:
It doesn't look like this function is used

Reply:
Removed